src/code/z_bgcheck.c

```diff
@@ -1,461 +1,4293 @@
+// ScaleRotPos equality test
+s32 func_8003E5B4(ScaleRotPos* a, ScaleRotPos* b) {
+    if (a->scale.x != b->scale.x || a->scale.y != b->scale.y || a->scale.z != b->scale.z || a->rot.x != b->rot.x ||
+        a->rot.y != b->rot.y || a->rot.z != b->rot.z || a->pos.x != b->pos.x || a->pos.y != b->pos.y ||
+        a->pos.z != b->pos.z) {
+        return false;
+    }
+    return true;
+}
+
+// Initialize DynaLookup lists
+void func_8003E688(DynaLookup* dynaLookup) {
+    func_8003871C(&dynaLookup->ceiling);
+    func_8003871C(&dynaLookup->wall);
+    func_8003871C(&dynaLookup->floor);
+}
+
+// Initialize DynaLookup
+void func_8003E6C4(DynaLookup* dynaLookup) {
+    dynaLookup->polyStartIndex = 0;
+    func_8003E688(dynaLookup);
+}
+
+// Init vtxStartIndex
+void func_8003E6E4(u16* vtxStartIndex) {
+    *vtxStartIndex = 0;
+}
+
+// Initialize BgActor
+void func_8003E6EC(GlobalContext* globalCtx, BgActor* bgActor) {
+    bgActor->actor = NULL;
+    bgActor->colHeader = NULL;
+    func_8003E530(&bgActor->prevSrp);
+    func_8003E530(&bgActor->curSrp);
+    func_8003E6C4(&bgActor->dynaLookup);
+    func_8003E6E4(&bgActor->vtxStartIndex);
+    bgActor->boundingSphere.center.x = bgActor->boundingSphere.center.y = bgActor->boundingSphere.center.z = 0;
+    bgActor->boundingSphere.radius = 0;
+}
+
+// setActor internal
+void func_8003E750(BgActor* bgActor, Actor* actor, CollisionHeader* colHeader) {
+    bgActor->actor = actor;
+    bgActor->colHeader = colHeader;
+    bgActor->prevSrp.scale = actor->scale;
+    bgActor->prevSrp.rot = actor->shape.rot;
+    bgActor->prevSrp.rot.x--;
+    bgActor->prevSrp.pos = actor->posRot.pos;
+    bgActor->curSrp.scale = actor->scale;
+    bgActor->curSrp.rot = actor->shape.rot;
+    bgActor->curSrp.pos = actor->posRot.pos;
+}
+
+// Test if BgActor has moved
+s32 func_8003E804(BgActor* bgActor) {
+    return func_8003E5B4(&bgActor->prevSrp, &bgActor->curSrp);
+}
+
+// NULL polyList
+void func_8003E82C(CollisionPoly** polyList) {
+    *polyList = NULL;
+}
+
+// Allocate dyna.polyList
+void func_8003E834(GlobalContext* globalCtx, CollisionPoly** polyList, s32 numPolys) {
+    *polyList = THA_AllocEndAlign(&globalCtx->state.tha, numPolys * sizeof(CollisionPoly), -2);
+    if (!(*polyList != NULL)) {
+        __assert("ptbl->pbuf != NULL", "../z_bgcheck.c", 6247);
+    }
+}
+
+// NULL vtxList
+void func_8003E888(Vec3s** vtxList) {
+    *vtxList = NULL;
+}
+
+// Allocate dyna.vtxList
+void func_8003E890(GlobalContext* globalCtx, Vec3s** vtxList, s32 numVtx) {
+    *vtxList = THA_AllocEndAlign(&globalCtx->state.tha, numVtx * sizeof(Vec3s), -2);
+    if (!(*vtxList != NULL)) {
+        __assert("ptbl->pbuf != NULL", "../z_bgcheck.c", 6277);
+    }
+}
+
+// Update BgActor's prevSrp
+void func_8003E8EC(GlobalContext* globalCtx, BgActor* bgActor) {
+    bgActor->prevSrp = bgActor->curSrp;
+}
+
+// Is BgActor
+s32 func_8003E934(s32 bgId) {
+    if (bgId < 0 || bgId >= BG_ACTOR_MAX) {
+        return false;
+    }
+    return true;
```

Comment:
the idea is that it's checking bgId to determine the type of bgId; hence Is BgActor tests if the id is a reference to a BgActor

Reply:
the type of bgId, as in dyna/static? then the implementation would have been `!= 50` not `>= 0 && < 50`

Reply:
yea i guess another way to think of the function is that it asks "is BgId referring to a dynaActor?"

It wouldn't make sense to implement it in that way, since the full range of ids that refer to BgActors is known to be 0 to BG_ACTOR_MAX-1. Additionally, -1 is used in a few locations in the code base.